src/GoogleAdWordsEventForwarder.js

```diff
@@ -124,23 +121,26 @@
             return 'Can\'t send to forwarder ' + name + ', not initialized. Event added to queue.';
         }
 
-        function setEnhancedConversionData(enhancedConversionData) {
-            if (enhancedConversionData.email) {
-                window.enhanced_conversion_data.email = enhancedConversionData.email;
-            }
-            if (enhancedConversionData.phone_number) {
-                window.enhanced_conversion_data.phone_number = enhancedConversionData.phone_number;
-            }
-            if (enhancedConversionData.first_name) {
-                window.enhanced_conversion_data.first_name = enhancedConversionData.first_name;
-            }
-            if (enhancedConversionData.last_name) {
-                window.enhanced_conversion_data.last_name = enhancedConversionData.last_name;
-            }
-            if (enhancedConversionData.home_address) {
-                window.enhanced_conversion_data.home_address =
-                    enhancedConversionData.home_address;
+        function hasEnhancedConversionData(customFlags) {
+            return customFlags && Object.keys(customFlags).length && customFlags[ENHANCED_CONVERSION_DATA];
+        }
+
+        function parseEnhancedConversionData(conversionData) {
+            // Checks if conversion data in custom flags is a stringified object
+            // Conversion data should only be a stringified object or JSON
+            if (typeof conversionData === 'string') {
+                try {
+                    return JSON.parse(conversionData);
+                } catch (error) {
+                    console.warn('Unrecognized Enhanced Conversion Data Format', conversionData, error);
+                    return {};
+                }
+            } else {
+                // Not a stringified object so it can be used as-is. However,
+                // we want to avoid mutating the original state, so we make a copy.
+                return cloneObject(conversionData);
```

Comment:
I'd put a try catch around here also, because there is nothing preventing them from passing a non-object, non-stringified object (ie, boolean, array, etc).  Then add a test for that.